### ISSUE TITLE
[TECH INSIGHTS] Time To Live Configuration not working as intended.

### ISSUE THREAD

**Issue by dayadev** [backstage/backstage#11595]
## Expected Behavior

Should clear the facts data in the database based on the configuration set using `lifecycle`

## Actual Behavior

Not clearing the facts

## Steps to Reproduce

I am trying to set lifecycle maxItems  to be kept in the database for each fact retriever/entity pair like below
createFactRetrieverRegistration({
        cadence: * * * * *,
        factRetriever: entityOwnershipFactRetriever,
        lifecycle: { maxItems: 1 },
      })

But this is not clearing the data as intended.

**Comment by jhaals** [comment]
@Xantier perhaps something you could help out investigating? 🙏

**Comment by Xantier** [comment]
This will need some more thorough investigation to figure out what seems to be issue. The likely problem comes from the query selecting facts, which is defined in `plugins/tech-insights-backend/src/service/persistence/TechInsightsDatabase.ts:213`.

The query produced by that ends up being like below, and should return the rows that should be deleted:
```
select *
from facts
       left join (select *
                  from (select id                                                                  as fid,
                               entity                                                              as fentity,
                               timestamp                                                           as ftimestamp,
                               row_number() over (partition by id, entity order by timestamp desc) as fact_rank
                        from facts) as ranks
                  where fact_rank <= :max_rows) as filterjoin
                 on filterjoin.fid = facts.id and filterjoin.fentity = facts.entity and
                    filterjoin.ftimestamp = facts.timestamp
where (id, entity) in (values (:fact_id, :entity_ref))
  and filterjoin.fid is null
```

I might have some time to look at this in the latter part of next week and see if we can mangle that into a better form.

**Comment by Xantier** [comment]
This issue ended up being more on the database level. When spinning up a new pg instance I noticed that the default precision for the timestamp column we are using to store fact rows is 6 digits. One of Knex/Luxon/JS in general doesn't support such a precise time correctly and therefore the eventual query fails to use that passed in value in the where clause properly. This can be mitigated on the DB config level, but I added a precision config to the column itself as well for untuned instances as well.
